### ISSUE TITLE
[Teaser] Saving without Alt Text with Image

### ISSUE THREAD

**Issue by juliaetate** [adobe/aem-core-wcm-components#2944]
## Bug Report

**Current Behavior**
In the Teaser component, if you have an image saved with alt text filled out, it is possible to delete the alt text and re-save an empty field without error. The field is marked as required but is not behaving as required. 

**Expected behavior/code**
Alt text should be required and behave as required when an Image is present whether it already existed or is newly uploaded.

**Environment**
- AEM SDK Version - 2025.5.21005.20250522T173058Z-250400
- Core Components version 2.29.0
- JRE version 11.0.27 2025-04-15 LTS

**Possible Solution**
The existing toggles for requiring the alt text don't update the marking for being required. Additionally, the required state doesn't seem to be available on load when an image is already present. 

**Additional context / Screenshots**

https://github.com/user-attachments/assets/ea220da0-470e-4884-b62f-0af9c0518666

**Comment by HitmanInWis** [comment]
Ran into this as well.

I believe the issue is in `/apps/core/wcm/components/image/v3/image/clientlibs/editor/js/image.js`

Code that determines whether alt text is required checks the following:

```
$("coral-fileupload.is-filled:not(:hidden)").length !== 0
```

However, when the Teaser dialog opens, the image fileupload field is hidden regardless of an image being set, since the Asset tab is not the first tab.

To fix this, I think the following code change can work:

```
$('coral-fileupload.is-filled').filter(function () {
    return this.style.display !== 'none';
}).length !== 0
```

That way the code will still return `true` if an image has been set and it's the correct image (based on the checkbox for using the page image or using the image set on the dialog) regardless of wether the tab is open.
